docs/source/examples.md

```diff
@@ -274,3 +274,87 @@ By default, the `hi-ml` package assumes that you supply a commandline argument `
 flag, coming out of your own argument parser, use the `submit_to_azureml` argument of the function
 `health.azure.himl.submit_to_azure_if_needed`. 
+
+## Training with k-fold cross validation in Azure ML
+To train with k-fold cross validation, you must do two things.
+
+1. Call the helper function `create_crossval_hyperdrive_config`
+to create an AML HyperDriveConfig object representing your parent run. It will have one child run for each of the k-fold
+splits you request, as follows
+    
+    ```python
+    from health_azure import create_crossval_hyperdrive_config
+    
+     hyperdrive_config = create_crossval_hyperdrive_config(num_cross_validation_splits,
+                                                           cross_val_split_name=cross_val_split_name,
+                                                           metric_name=metric_name)
+    ```
+    where:
+    - `num_cross_validation_splits` is the number of k-fold cross validation splits you require
+    - `cross_val_split_name` is the name of the argument given to each child run, whose value denotes which split that child represents (this parameter
+    defaults to 'cross_validation_split_index', in which case, supposing you specified 2 cross validation splits, one would 
+    receive the arguments ['cross_validation_split_index' '0'] and the other would receive ['cross_validation_split_index' '1']].
+    It is up to you to then use these args to retrieve the correct split from your data.
+    - `metrics_name` represents the name of a metric that you will compare your child runs by. **NOTE** the
+    run will expect to find this metric, otherwise it will fail [as described here](
+    https://docs.microsoft.com/en-us/azure/machine-learning/how-to-tune-hyperparameters#log-metrics-for-hyperparameter-tuning
+    )
+    You can log this metric in your training script as follows:
+     ```python
+    from azureml.core import Run
+    
+    # Example of logging a metric called <metric_name> to an AML Run.
+    loss = <my_loss_calc>
+    run_log = Run.get_context()
+    run_log.log(metric_name, loss)
+    ```
+   See the [documentation here](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-log-view-metrics) for
+   further explanation.
+
+2. The hyperdrive_config returned above must be passed into the function `submit_to_azure_if_needed` as follows:
+    
+    ```python
+     run_info = submit_to_azure_if_needed(
+            ...
+            hyperdrive_config=hyperdrive_config
+    )
+    ```
+    This will create a parent (HyperDrive) Run, with `num_cross_validation_split` children - each one associated with a different data split.
+
+## Retrieving the aggregated results of a cross validation/ HyperDrive run
+
+You can retrieve a Pandas DataFrame of the aggregated results from your cross validation run as follows: 
+
+```python
+from health_azure import utils
+
+df = utils.aggregate_hyperdrive_metrics(run.id, workspace)
```

Comment:
can we have a more friendly overload that accepts a string runID, and auto fills in the workspace?